DataFormats/Common/src/ProductData.cc

```diff
@@ -38,4 +38,8 @@ namespace edm {
   void ProductData::unsafe_setWrapper(std::unique_ptr<WrapperBase> iValue) const {
     wrapper_ = std::move(iValue);
   }
+
+  void ProductData::unsafe_setWrapper(std::shared_ptr<WrapperBase const> iValue) const {
+    wrapper_ = std::const_pointer_cast<WrapperBase>(std::move(iValue));
```

Comment:
Is it possible to just make `wrapper_` a `std::shared_ptr<const WrapperBase>` ?

Reply:
In principle yes, I suppose. But I believe that would mean adding `const_cast` to
https://github.com/cms-sw/cmssw/blob/a480cc2041dd20d6ebd2ec67eb050b055f90f728/DataFormats/Common/interface/ProductData.h#L36
, or deal with "modifying a `const` object" in
https://github.com/cms-sw/cmssw/blob/a480cc2041dd20d6ebd2ec67eb050b055f90f728/FWCore/Framework/src/ProductResolvers.cc#L93-L110
or
https://github.com/cms-sw/cmssw/blob/a480cc2041dd20d6ebd2ec67eb050b055f90f728/DataFormats/Common/interface/WrapperDetail.h#L98-L102